--- apmd-3.2.2.orig/Makefile.lib64	2005-02-11 04:37:38.825303788 -0500
+++ apmd-3.2.2.orig/Makefile	2005-02-11 04:37:50.555670055 -0500
@@ -26,6 +26,12 @@ ifneq (,)
 This makefile requires GNU Make.
 endif
 
+include config.make
+
+ifeq ($(LIB),)
+LIB = lib
+endif
+
 .SUFFIXES:
 
 OBJS=apmlib.o
@@ -36,7 +42,7 @@ PREFIX=/usr
 MANDIR=$(PREFIX)/man
 BINDIR=$(PREFIX)/bin
 SBINDIR=$(PREFIX)/sbin
-LIBDIR=$(PREFIX)/lib
+LIBDIR=$(PREFIX)/$(LIB)
 INCDIR=$(PREFIX)/include
 PROXY_DIR=/etc
 DESTDIR=
@@ -50,7 +56,7 @@ XTRACFLAGS=-Wall -pipe -I. -I/usr/src/li
 LDFLAGS=
 LIBAPM=libapm.la
 LIBAPM_FLAGS=-rpath $(LIBDIR) -version-info $(LIB_VERSION)
-XLDFLAGS=-L/usr/X11R6/lib
+XLDFLAGS=-L/usr/X11R6/$(LIB)
 XLIBS=-lXaw -lXmu -lXt -lXext -lSM -lICE -lX11
 RANLIB=ranlib
 
